src/cascadia/Remoting/MonarchFactory.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+
+#include "Monarch.h"
+
+// This seems like a hack, but it works.
+//
+// This class factory works so that there's only ever one instance of a Monarch
+// per-process. Once the first monarch is created, we'll stash it in g_weak.
+// Future callers who try to instantiate a Monarch will get the one that's
+// already been made.
+
+winrt::weak_ref<winrt::Microsoft::Terminal::Remoting::implementation::Monarch> g_weak{ nullptr };

```

Comment:
var storage locations in a header == bad news. importing the header multiple times (even accidentally) will cause linking failures

Reply:
make it a static inside the CreateInstance method perhaps